src/components/proofs/EmailProof.tsx

```diff
@@ -81,12 +81,29 @@ const questionBlock = (open: boolean) =>
 
 const tooltipWrapper = classnames(display('flex'), flex('flex-1'))
 
-export default function () {
-  const [domain, setDomain] = useState('')
-  const [open, setOpen] = useState(false)
+export default function ({
+  domain: defaultDomain,
+  token: defaultToken,
+}: {
+  domain?: string
+  token?: string
+}) {
+  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [domain, setDomain] = useState(defaultDomain ?? '')
+  const [open, setOpen] = useState(!!defaultDomain)
   const [error, setError] = useState<string | undefined>()
   const [generationStarted, setGenerationStarted] = useState(false)
-  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [token, setToken] = useState(defaultToken ?? '')
+
+  useEffect(() => {
+    if (defaultToken && defaultDomain) {
```

Comment:
`if (!defaultToken || !defaultDomain) return`
Less nesting = better code, usually, let's keep it

Reply:
Changed it too